### ISSUE TITLE
Simultaneous URL input and dropdown select

### ISSUE THREAD

**Issue by lambdadelta34** [swagger-api/swagger-ui#4292]
Hi, is it possible to have both input field for URL and options list with pre-defined urls attribute in topbar?

**Comment by shockey** [comment]
Not currently - the presence of a `urls` parameter causes the URL input to be replaced with a drop down list.

PRs that add this functionality are welcome 😄

**Comment by heldersepu** [comment]
@lambdadelta34 If you want someone to consider this you should leave it open...
and perhaps improve a bit the details...

The HTML5 `<datalist>` Tag could be what you are looking for
https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_datalist

**Comment by lambdadelta34** [comment]
@heldersepu OK, I'll leave issue opened. Though not sure about details, `<datalist>` tag looks like exactly what i wanted, and i'll try to find time to make PR.

**Comment by heldersepu** [comment]
about the details, set some expectations:
- how do you think it should look
- how should it behave

Maybe add a couple of screenshots with rough mockups

**Comment by erajkovic** [comment]
Yes, it would be nice to be able to enter a different URL that the default one, so that we can test using distinct remote endpoints, which may not be in the original set at the time the spec was published. I have been struggling with a test swagger (open api 3) file, that was using this snippet:
  "servers": Array[1][
    {
      "url": "http://example.com/",
      "variables": {
        
      }
    }
  ],
one work around is to go update the spec like below, but it will be nice to be able to enter a new URL after publishing 
  "servers": Array[1][
    {
      "url": "https://apicatalog.us.oracle.com/",
      "variables": {
        
      }
    }
  ],

We used to to it outside the Swagger UI in v2

![apic-swagger-server-list](https://user-images.githubusercontent.com/4298647/37377492-82b814cc-26e7-11e8-8ea6-2fdd92b36e3b.png)

test URL (https://apicatalog.oraclecloud.com/ui/views/apicollection/oracle-public/database/csdbr-13 - need to authenticate)

**Comment by erajkovic** [comment]
It would also be useful to be able to enter a URL when there is no server defined in the spec. Maybe it could be driven by a flag to enable-edit-server property of some kind - not sure what a good name can be, to align with the current naming scheme.

**Comment by webron** [comment]
@erajkovic - your comment is talking about a different feature than the requested by the OP. The OP is referring to the URL(s) that's used to load the API definition, whereas your request is regarding the `servers` section in a given API definition. If you'd like that feature considered, please file a separate ticket.

**Comment by erajkovic** [comment]
When I first looked at the 'servers' URLs, I found that it was replacing/ignoring the default URL value, so we decided to continue to use the default URL and avoid using any servers URLs in our setup. Can rewrite this as a separate issue if you would like it so.

**Comment by erajkovic** [comment]
done - see issue 4735

**Comment by shockey** [comment]
@erajkovic thanks for filing! I'll take a look.

Also - I'm going to minimize our discussion here, to keep this thread focused on the original issue.
